[opentelemetry-operator] Remove kube-rbac-proxy and use built-in metrics auth

YASHMAHAKAL opened the pull request:
## Description
Removes kube-rbac-proxy dependency and migrates to controller-runtime's built-in metrics authentication, aligning with upstream operator v0.142.0+ changes.

## Changes
- Removed kube-rbac-proxy sidecar container
- Replaced `kubeRBACProxy` config with `metricsServer` config
- Added support for `--metrics-secure` flag and TLS options
- Updated RBAC to remove proxy-specific permissions
- Simplified service to single metrics port
- Chart version bumped to 0.103.0

## Migration
Default behavior unchanged (metrics on port 8080). For secure metrics, set `metricsServer.secure: true`, as described in UPGRADING.md.

## Testing
Helm lint passes
Templates render correctly
Examples regenerated
Tested on local cluster (Minikube)

Fixes #1999

YASHMAHAKAL committed:
Remove kube-rbac-proxy and use built-in metrics auth

Signed-off-by: YASHMAHAKAL <yvsst01@gmail.com>

TylerHelmuth commented:
Switching to draft until CoPilot is done reviewing.

YASHMAHAKAL commented:
@TylerHelmuth Copilot review is complete. I've already addressed the port default issue (commit d9931e6). The remaining comments are style suggestions. Should I address them all, or would you like to review first? because i'm not much sure as first two are regarding UPGRADING.md...

TylerHelmuth commented:
@YASHMAHAKAL close any copilot reviews you're ignoring and then mark the PR ready for review when you want us to review it. I dont want to read through copilot comments.

YASHMAHAKAL committed:
fix: remove trailing spaces in values.yaml

Signed-off-by: YASHMAHAKAL <yvsst01@gmail.com>

jaronoff97 reviewed (approved):
Should be good @pavolloffay mind taking a look?

rogercoll reviewed (commented):
Should we bump `opentelemetry-collector-k8s` image to `0.142.0` as well? https://github.com/open-telemetry/opentelemetry-helm-charts/pull/2004/files#diff-56c52957e9b80d7dcb9b9771e329d7dc4f1a3ee8fb22acdcb3a75c1c92e75c18R53

jaronoff97 commented:
@rogercoll yes i think we should @YASHMAHAKAL apologies one more change

YASHMAHAKAL commented:
> @rogercoll yes i think we should @YASHMAHAKAL apologies one more change

Thanks @jaronoff97, it's absolutely okay ! i'll be updating the PR with necessary changes : )

TylerHelmuth commented:
@YASHMAHAKAL looks like CI is still failing

YASHMAHAKAL committed:
fix: add missing newline to values.schema.json

Signed-off-by: YASHMAHAKAL <yvsst01@gmail.com>

YASHMAHAKAL commented:
@TylerHelmuth, Thanks for your patience.. i've made a fix, please review it : )

jaronoff97 reviewed (commented):
@YASHMAHAKAL PR looks good, but youll need to bump the Chart version. Given this is technically a breaking change, please bump the minor version instead of the patch version

swiatekm commented:
@YASHMAHAKAL you'll need to revert the changes from https://github.com/open-telemetry/opentelemetry-helm-charts/pull/2057/changes/6c98bba5e87aa92642fb7125329e10dad061c594 to get the e2e tests to pass.

YASHMAHAKAL committed:
chore: bump version to 0.106.0 and fix E2E tests

Remove invalid kubeRBACProxy configuration from workflow.
Regenerate examples with new version labels.

Signed-off-by: YASHMAHAKAL <yvsst01@gmail.com>

YASHMAHAKAL commented on charts/opentelemetry-operator/examples/default/rendered/tests/test-service-connection.yaml:
```diff
@@ -59,7 +59,7 @@ metadata:
   namespace: default
   labels:
-    helm.sh/chart: opentelemetry-operator-0.105.0
+    helm.sh/chart: opentelemetry-operator-0.106.0
```
@jaronoff97 minor version bumped to 0.106.0 as per your recommendation, Thanks !

swiatekm commented on charts/opentelemetry-operator/values.yaml:
```diff
@@ -233,38 +233,25 @@ manager:
+
+  # Port for the metrics server
+  # Use 8443 when secure is true, 8080 when false
+  port: 8443
```
Can we not use `manager.ports.metrics` instead? That would remove a lot of the conditional logic we have around ports and we wouldn't have two sources of truth for them. The HTTPS settings could go into a separate section, like `manager.metrics`.

swiatekm reviewed (changes requested):
I'd prefer if we put the new config section under `manager` and only had one way of setting the metrics port. I've left some comments with suggestions.

Notably, if we're enabling HTTPS by default, we also need to deal with https://github.com/open-telemetry/opentelemetry-helm-charts/issues/2063 by configuring the ServiceMonitor appropriately.

jaronoff97 commented on charts/opentelemetry-operator/values.yaml:
```diff
@@ -233,38 +233,25 @@ manager:
+## Metrics server configuration
+## The operator uses controller-runtime built-in auth for the metrics server
+## Authentication is disabled by default for backward compatibility
+metricsServer:
```
this makes sense to me.

eenchev commented:
Hi @swiatekm @YASHMAHAKAL , do you think anything can be done re this PR? Essentially, the chart has no proper way to configure secure metrics collection on the newer versions of operator. So there is discrepency between the chart configuration and the actual operator until this is merged.
